### PR TITLE
[example] Add order-independent transparency example

### PR DESCRIPTION
**Pull request by lin-hitonami**
Issue: #5420

### Brief Summary
image

### CODE REVIEW

File: python/taichi/examples/rendering/oit_renderer.py [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+                vec3(ti.random() * 4 - 2,
+                     ti.random() * 4 - 2,
+                     ti.random() * 3 - 1.5),
+                ti.random() * 0.3 + 0.1,
+                (ti.random(), ti.random(), ti.random(),
+                 ti.random() * 0.3 + 0.2)))
+
+    for u, v in color_buffer:
+        aspect_ratio = res[0] / res[1]
+        pos = camera_pos
```
Comment (neozhaoliang) [inline python/taichi/examples/rendering/oit_renderer.py:137]: `aspect_ratio` can be made global, and `pos` is not used later?

File: python/taichi/examples/rendering/oit_renderer.py [hunk 2]
```diff
@@ -0,0 +1,157 @@
+import taichi as ti
+from taichi.math import clamp, mix, vec3, vec4
```
Comment (neozhaoliang) [inline python/taichi/examples/rendering/oit_renderer.py:2]:
```suggestion
from taichi.math import clamp, mix, vec3, vec4, normalize
```

File: python/taichi/examples/rendering/oit_renderer.py [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+            t1 = tp - tt
+            if t1 > 0:
+                hit_pos1 = light.pos + light.dir * t1
+                dist1 = t1
+                normal1 = (hit_pos1 - sphere.center).normalized()
+            t2 = tp + tt
+            if t2 > 0:
+                hit_pos2 = light.pos + light.dir * t2
+                dist2 = t2
+                normal2 = (hit_pos2 - sphere.center).normalized()
```
Comment (neozhaoliang) [inline python/taichi/examples/rendering/oit_renderer.py:78]:
```suggestion
                normal1 = (hit_pos1 - sphere.center).normalized()
            t2 = tp + tt
            if t2 > 0:
                hit_pos2 = light.pos + light.dir * t2
                dist2 = t2
                normal2 = normalize(hit_pos2 - sphere.center)
```

File: python/taichi/examples/rendering/oit_renderer.py [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+                ti.random() * 0.2 + 0.1,
+                (ti.random(), ti.random(), ti.random(),
+                 ti.random() * 0.3 + 0.2)))
+
+    for u, v in color_buffer:
+        ray_dir = ti.Vector([
+            (2 * fov * u / res[1] - fov * aspect_ratio - 1e-5),
+            (2 * fov * v / res[1] - fov - 1e-5),
+            -1.0,
+        ])
```
Comment (neozhaoliang) [inline python/taichi/examples/rendering/oit_renderer.py:141]:
```suggestion
        ray_dir = vec3(
            (2 * fov * u / res[1] - fov * aspect_ratio - 1e-5),
            (2 * fov * v / res[1] - fov - 1e-5),
            -1.0
        )
```

File: python/taichi/examples/rendering/oit_renderer.py [hunk 5]
```diff
@@ -0,0 +1,157 @@
...
+        ray_dir = ray_dir.normalized()
```
Comment (neozhaoliang) [inline python/taichi/examples/rendering/oit_renderer.py:144]:
```suggestion
        ray_dir = normalize(ray_dir)
```

**Review by yuanming-hu — APPROVED** [review]
Very cool!

File: python/taichi/examples/rendering/oit_renderer.py [hunk 6]
```diff
@@ -70,12 +70,12 @@ def intersect_sphere(light: ti.template(), sphere: ti.template()):
             if t1 > 0:
                 hit_pos1 = light.pos + light.dir * t1
                 dist1 = t1
-                normal1 = (hit_pos1 - sphere.center).normalized()
+                normal1 = normalize((hit_pos1 - sphere.center)
```
Comment (neozhaoliang) [inline python/taichi/examples/rendering/oit_renderer.py]:
```suggestion
                normal1 = normalize(hit_pos1 - sphere.center)
```
I'm sorry this '(' was not closed...

Comment (lin-hitonami) [reply python/taichi/examples/rendering/oit_renderer.py]: fixed

**Review by neozhaoliang — APPROVED** [review]
LGTM!

**Pull request merged**